### ISSUE TITLE
language or at least input language

### ISSUE THREAD

**Issue by sabaak95** [sapplica/sentrifugo#139]
hi
i wanted to know if i can change the language of menus ( i mean just the appearance of the menus and what it shows to users ) 
if not possible, i wanted to know a way to change the input . now whenever i want to add a candidate or a position , when entering the names , the program doesnt allow me to add input in other languages . i wanted to know where i can change this in the codes .

**Comment by ch-gz** [comment]
The project is not support multi language for now, may be a feature in the future. BTW, you can change those menus and interface language by changing those view scripts in source code.

**Comment by sabaak95** [comment]
@ch-gz  i just want application to save input for interview comments , interview feedback and candidates names in persian ! nothing more

**Comment by ch-gz** [comment]
@sabak95 Do you mean when you add candidate, the name field won't allow you to enter any other special character(Like "Clàudia" or "Mònica")? If this is what  you want , it's very easy to fix.

find file in your project"application/modules/default/forms/Candidatedetails.php", delete or comment from `line 73-78, line 86-91.` 
append these two line into your "application\configs\application.ini"
```
resources.view.encoding = "UTF-8"
resources.db.params.charset = "utf8"
```
Now most pages will display the right encoding.

**Comment by sabaak95** [comment]
@ch-gz Thank you like a million times :) . as i said before , i also want this to be ok for interview feedback and comments. after applying the change to only the candidatedetails.php , the problem was solved for interview comments too . but i wanted to know if it is neccessary to apply this change to other files related to interviews like interviewrounds.php . because they also had something like addvalidator in line 217-228

**Comment by ch-gz** [comment]
@sabak95 They are using REGEX to filter user input ,you can remove it so anything you can type in that field. Or you can change the rule you want. Actually, to filter user input is kind of necessary.

**Comment by klukiyan** [comment]
@ch-gz  your comment with editing application.ini is amaizing.
It fixed Czech characters display on all pages that appeared broken

Do you have similar hint for getting the czech names/surnames also appear in UTF-8
resources.view.encoding = "UTF-8"
resources.db.params.charset = "utf8"

**Comment by ch-gz** [comment]
@klukiyan Czech names will display in UTF8 encoding. Just make sure database set to UTF8 for columns and also database connection. FYI [Encoding UTF-8 for Czech chars](https://stackoverflow.com/questions/20688512/encoding-utf-8-for-czech-chars)

**Comment by klukiyan** [comment]
@ch-gz thank you very much for your response.
I figured out how to fix the email encoding yesterday. The Sentrifugo tool uses the PHPMailer module
`\application\modules\default\library\PHPMailer\class.phpmailer.php`
On line 47 there was encoding to some ISO format (I believe it's some latin1)
I changed it to utf-8 and Czech characters started displaying properly in emails

**Comment by ch-gz** [comment]
@klukiyan Yes, they are using PHPMailer+Templates(application/modules/default/view/scripts/mailtemplates) to send email. You can also change the mail templates.

**Comment by klukiyan** [comment]
@ch-gz  Hi Gab,
Can you please help me again. Everywhere the Czech characters appear ok, except in the tables that the tool generates in /index.php/manageremployeevacations/
It shows weird symbols instead. I assume it's some hidden component that has a broken charset, but I have no idea how to find it. It's really the last place where I need to fix the czech characters to display well and can't ...

I researched more, and it looks like everywhere there is this table element that can be sorted, etc... the characters don't display well. It just visibly impacts names only.
Something wrong with this "popup helper". Don't know how to fix it and force it to utf8 as everywhere else
`<?php echo sapp_Helper::popup_helper('leaverequest');?>
`
_**application\modules\default\views\scripts\manageremployeevacations\index.phtml**_
@sapplica can you help?

**Comment by PetelsonLuis** [comment]
HI, how are you. 
If you want change menu language go to database in tables main_menu and edit rows.

**Comment by hoangnguyenbao** [comment]
> @ch-gz Hi Gab,
> Can you please help me again. Everywhere the Czech characters appear ok, except in the tables that the tool generates in /index.php/manageremployeevacations/
> It shows weird symbols instead. I assume it's some hidden component that has a broken charset, but I have no idea how to find it. It's really the last place where I need to fix the czech characters to display well and can't ...
> 
> I researched more, and it looks like everywhere there is this table element that can be sorted, etc... the characters don't display well. It just visibly impacts names only.
> Something wrong with this "popup helper". Don't know how to fix it and force it to utf8 as everywhere else
> `<?php echo sapp_Helper::popup_helper('leaverequest');?> `
> _**application\modules\default\views\scripts\manageremployeevacations\index.phtml**_
> @sapplica can you help?

I have the same issue. Please some one can help me? For example: website show "Nguyá»�n Thá»� HÃ  Linh", but when I point at that to show title or click View detail, text will display well: "Nguyễn Thị Hà Linh".

**Comment by ch-gz** [comment]
@hoangnguyenbao @klukiyan It looks like the database tables not format as UTF8. BTW I've no longer use sentrifugo for a while, so I cannot reproduce the error.

**Comment by luaikayyali** [comment]
Hello,
Those interested in changing the language to Arabic, instead of deleting the lines 73-78, line 86-91, you can change the Regex pattern into this:-
'pattern'=>'/^[a-zA-Zأبجدهـ،وزحطيكلمنسعفصقرشتاىؤثذدخغضطظةءئ.\- ?]+$/',
this line is 87 and 74

As of those who need Spanish for example, you can just enter the Spanish characters after the letter Z to the dot (.)

Enjoy.

By the way, I managed to change the interface to Arabic, it was very hectic, as the labels are scattered everywhere in the system, unfortunately, there is no consistency. But overall the portal is nice and can be used.

Regards,
Tigershark

**Comment by Kamalallahabi** [comment]
> 
> 
> Hello,
> Those interested in changing the language to Arabic, instead of deleting the lines 73-78, line 86-91, you can change the Regex pattern into this:-
> 'pattern'=>'/^[a-zA-Zأبجدهـ،وزحطيكلمنسعفصقرشتاىؤثذدخغضطظةءئ.- ?]+$/',
> this line is 87 and 74
> 
> As of those who need Spanish for example, you can just enter the Spanish characters after the letter Z to the dot (.)
> 
> Enjoy.
> 
> By the way, I managed to change the interface to Arabic, it was very hectic, as the labels are scattered everywhere in the system, unfortunately, there is no consistency. But overall the portal is nice and can be used.
> 
> Regards,
> Tigershark

Hello brother
can you help me
I didn't get Arabic how do I get it
